### ISSUE TITLE
Thread 'main' panicked under i3wm

### ISSUE THREAD

**Issue by Sonico98** [pierrechevalier83/workstyle#14]
First, thanks for implementing #6!
It seems to work fine under sway, but I'm an i3wm user so maybe someone else can comment more on that.

Under i3wm-gaps, trying to start workstyle 3.0 returns an error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unknown variant `dockarea`, expected one of `root`, `output`, `workspace`, `con`, `floating_con`", line: 1, column: 6170)', /home/sonico/.cargo/registry/src/github.com-1ecc6299db9ec823/workstyle-0.3.0/src/main.rs:159:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Running it with RUST_BACKTRACE=1 returns:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unknown variant `dockarea`, expected one of `root`, `output`, `workspace`, `con`, `floating_con`", line: 1, column: 6167)

   0: failure::backtrace::internal::InternalBacktrace::new
   1: failure::backtrace::Backtrace::new
   2: swayipc::sync::connection::Connection::get_tree
   3: workstyle::main
   4: std::sys_common::backtrace::__rust_begin_short_backtrace
   5: std::rt::lang_start::{{closure}}
   6: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/1.56.1/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at /rustc/1.56.1/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/1.56.1/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/1.56.1/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/1.56.1/library/std/src/rt.rs:45:48
      std::panicking::try::do_call
             at /rustc/1.56.1/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/1.56.1/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/1.56.1/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal
             at /rustc/1.56.1/library/std/src/rt.rs:45:20
   7: main
   8: __libc_start_main
   9: _start
', /home/sonico/.cargo/registry/src/github.com-1ecc6299db9ec823/workstyle-0.3.0/src/main.rs:159:34
stack backtrace:
   0: rust_begin_unwind
             at /rustc/1.56.1/library/std/src/panicking.rs:517:5
   1: core::panicking::panic_fmt
             at /rustc/1.56.1/library/core/src/panicking.rs:101:14
   2: core::result::unwrap_failed
             at /rustc/1.56.1/library/core/src/result.rs:1617:5
   3: workstyle::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
And with RUST_BACKTRACE=full:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unknown variant `dockarea`, expected one of `root`, `output`, `workspace`, `con`, `floating_con`", line: 1, column: 6167)

   0: failure::backtrace::internal::InternalBacktrace::new
   1: failure::backtrace::Backtrace::new
   2: swayipc::sync::connection::Connection::get_tree
   3: workstyle::main
   4: std::sys_common::backtrace::__rust_begin_short_backtrace
   5: std::rt::lang_start::{{closure}}
   6: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/1.56.1/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at /rustc/1.56.1/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/1.56.1/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/1.56.1/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/1.56.1/library/std/src/rt.rs:45:48
      std::panicking::try::do_call
             at /rustc/1.56.1/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/1.56.1/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/1.56.1/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal
             at /rustc/1.56.1/library/std/src/rt.rs:45:20
   7: main
   8: __libc_start_main
   9: _start
', /home/sonico/.cargo/registry/src/github.com-1ecc6299db9ec823/workstyle-0.3.0/src/main.rs:159:34
stack backtrace:
   0:     0x5626f592dd0c - std::backtrace_rs::backtrace::libunwind::trace::h94b503c36dba7903
                               at /rustc/1.56.1/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x5626f592dd0c - std::backtrace_rs::backtrace::trace_unsynchronized::h655a42dea680a248
                               at /rustc/1.56.1/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x5626f592dd0c - std::sys_common::backtrace::_print_fmt::h4110bbd76bdd1b8a
                               at /rustc/1.56.1/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x5626f592dd0c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb300d7b4bcb7abbc
                               at /rustc/1.56.1/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x5626f595221c - core::fmt::write::h872e832230b6cc2d
                               at /rustc/1.56.1/library/core/src/fmt/mod.rs:1150:17
   5:     0x5626f592a685 - std::io::Write::write_fmt::h2def0cb416325237
                               at /rustc/1.56.1/library/std/src/io/mod.rs:1667:15
   6:     0x5626f592f680 - std::sys_common::backtrace::_print::ha4a4bdfdc780db67
                               at /rustc/1.56.1/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x5626f592f680 - std::sys_common::backtrace::print::h400e21d23f1f70b4
                               at /rustc/1.56.1/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x5626f592f680 - std::panicking::default_hook::{{closure}}::h23ed76f29cbcf4c5
                               at /rustc/1.56.1/library/std/src/panicking.rs:210:50
   9:     0x5626f592f237 - std::panicking::default_hook::h2cf73a02d82f5d12
                               at /rustc/1.56.1/library/std/src/panicking.rs:227:9
  10:     0x5626f592fd34 - std::panicking::rust_panic_with_hook::hcfb1400d4b55f82b
                               at /rustc/1.56.1/library/std/src/panicking.rs:624:17
  11:     0x5626f592f810 - std::panicking::begin_panic_handler::{{closure}}::hebf77047a5d73716
                               at /rustc/1.56.1/library/std/src/panicking.rs:521:13
  12:     0x5626f592e1d4 - std::sys_common::backtrace::__rust_end_short_backtrace::hd843165880920e54
                               at /rustc/1.56.1/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x5626f592f779 - rust_begin_unwind
                               at /rustc/1.56.1/library/std/src/panicking.rs:517:5
  14:     0x5626f57b3e21 - core::panicking::panic_fmt::h3c031ae033919cfd
                               at /rustc/1.56.1/library/core/src/panicking.rs:101:14
  15:     0x5626f57b3f13 - core::result::unwrap_failed::h2008e92911cd236c
                               at /rustc/1.56.1/library/core/src/result.rs:1617:5
  16:     0x5626f57c885b - workstyle::main::hddc6550541ff12b1
  17:     0x5626f57c8ec3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h67b669155d72f518
  18:     0x5626f57c0909 - std::rt::lang_start::{{closure}}::h83247b39865352f7
  19:     0x5626f593023a - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hb321482de4b67141
                               at /rustc/1.56.1/library/core/src/ops/function.rs:259:13
  20:     0x5626f593023a - std::panicking::try::do_call::h3fed621b0fa24aca
                               at /rustc/1.56.1/library/std/src/panicking.rs:403:40
  21:     0x5626f593023a - std::panicking::try::h204b4d69d6643943
                               at /rustc/1.56.1/library/std/src/panicking.rs:367:19
  22:     0x5626f593023a - std::panic::catch_unwind::h0a51b863b254eb5d
                               at /rustc/1.56.1/library/std/src/panic.rs:129:14
  23:     0x5626f593023a - std::rt::lang_start_internal::{{closure}}::hde35fcb03bef19dc
                               at /rustc/1.56.1/library/std/src/rt.rs:45:48
  24:     0x5626f593023a - std::panicking::try::do_call::h01394369af4fdc18
                               at /rustc/1.56.1/library/std/src/panicking.rs:403:40
  25:     0x5626f593023a - std::panicking::try::hc8bf72467451e254
                               at /rustc/1.56.1/library/std/src/panicking.rs:367:19
  26:     0x5626f593023a - std::panic::catch_unwind::h6939059e753a76c5
                               at /rustc/1.56.1/library/std/src/panic.rs:129:14
  27:     0x5626f593023a - std::rt::lang_start_internal::h176b3570df33ee75
                               at /rustc/1.56.1/library/std/src/rt.rs:45:20
  28:     0x5626f57c8ad2 - main
  29:     0x7f5943037b25 - __libc_start_main
  30:     0x5626f57b45fe - _start
  31:                0x0 - <unknown>
  ```

I've installed workstyle using cargo, and I'm running under Arch Linux. If any other details are needed, please tell me.
Under sway there are no errors, and the previous version worked fine under i3.
And sorry for not being able to contribute with code!

**Comment by pierrechevalier83** [comment]
Oops! Thanks so much for the bug report, @Sonico98 . I will install i3 and look into it :)

**Comment by pierrechevalier83** [comment]
I have reproduced the error locally with i3 and I understand what is going on: I switched the backend of this crate to `swayipc`, naively thinking that that crate provider compatibility with both sway and i3, but when I try to call `get_tree`, `swayipc` fails because one of the nodes has type "dockarea" which is i3 specific.

What I will need to do is implement a backend that is compatible with both i3 and sway, falling back to `i3ipc` when i3 is the running window manager.

I am on it. It may be slightly non-trivial and take me a short while.
